File: src/database/seeders/RoleSeeder.php
<?php

namespace Database\Seeders;

use App\Models\Role;
use Illuminate\Database\Seeder;

class RoleSeeder extends Seeder
{
    /**
     * Run the database seeds.
     */
    public function run(): void
    {
        $roles = [
            [
                'name' => 'Admin',
                'description' => 'Administrador del sistema con acceso completo',
            ],
            [
                'name' => 'Operador',
                'description' => 'Usuario con permisos de creación y edición justos para cumplir sus tareas',
            ],
            [
                'name' => 'Administracion',
                'description' => 'Usuario con permisos de creación y edición, mas involucrado en operactiones de administracion',
            ],
            [
                'name' => 'Consultor',
                'description' => 'Consultor con acceso de solo lectura',
            ],
        ];

        foreach ($roles as $role) {
            Role::create($role);
        }
    }
}

File: backend-dotnet/src/Ikhana.Infrastructure/Data/Seeders/RoleSeeder.cs
using Ikhana.Application.Common.Interfaces;
using Ikhana.Domain.Entities;
using Microsoft.AspNetCore.Identity;

namespace Ikhana.Infrastructure.Data.Seeders;

public class RoleSeeder : IDataSeeder
{
    private readonly RoleManager<ApplicationRole> _roleManager;

    public int Order => 1;

    public RoleSeeder(RoleManager<ApplicationRole> roleManager)
    {
        _roleManager = roleManager;
    }

    public async Task SeedAsync(CancellationToken cancellationToken = default)
    {
        var roles = new (string Name, string Description)[]
        {
            ("Admin", "Administrador del sistema con acceso completo"),
            ("Operador", "Usuario con permisos de creacion y edicion justos para cumplir sus tareas"),
            ("Administracion", "Usuario con permisos de creacion y edicion, mas involucrado en operactiones de administracion"),
            ("Consultor", "Consultor con acceso de solo lectura"),
        };

        foreach (var (name, description) in roles)
        {
            if (await _roleManager.FindByNameAsync(name) is not null)
                continue;

            var role = new ApplicationRole
            {
                Name = name,
                Description = description,
                ConcurrencyStamp = Guid.NewGuid().ToString(),
            };

            await _roleManager.CreateAsync(role);
        }
    }
}
